css-transitions/styles.css

```diff
@@ -0,0 +1,56 @@
+* {
+  box-sizing: border-box;
+}
+
+body {
+  margin: 0;
+  font-family: sans-serif;
+  background-color: rgb(245, 245, 245);
+}
+
+section {
+  max-width: 992px;
+  margin: 2rem auto;
+  display: flex;
+  justify-content: space-evenly;
+}
+
+.custom-button {
+  padding: 0.5rem 1.5rem;
+  border: none;
+  border-radius: 0.25rem;
+  font-size: 1rem;
+  color: white;
+  background-color: blue;
+}
+
+.custom-button:hover {
+  background-color: red;
+  transition: 0.5s;
```

Comment:
The `transition` property should be applied to the main rule for an element, not the rule including the pseudoclass. If you only apply it to the pseudoclass, your transition will only work while the event is occurring, as soon as the event pseudoclass is removed, the transition would disappear as well.